### PR TITLE
refactor: versioning with cmake

### PR DESCRIPTION
**Pull request by leogr**
**What type of PR is this?**

/kind design

/kind documentation

/kind feature

**Any specific area of the project related to this PR?**

> /area CI

**Does this PR require a change in the driver versions?**
No.

**What this PR does / why we need it**:

This PR improves the build system's ability to automatically construct version numbers string, in particular for development builds. The new implementation ensures that the resulting strings are consistent with the SemVer specification and can be ordered in the proper way. More detail about the revised convention and implementation can be found in the updated README.md by this PR. As a bonus point, this PR also removes ad unnecessary dependency in the spirit of streamlining our 3rd-party dependencies.

Once this PR gets merged and libs bumped in Falco, the new cmake module and the revised convention can also be used by Falco.

/milestone 0.13.0

**Does this PR introduce a user-facing change?**:

```release-note
new: cmake module to handle released and development build versioning numbers
```

### CODE REVIEW

File: cmake/modules/GetVersionFromGit.cmake [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+        endif()
+
+        _falcosecurity_execute_git(tag describe --tags ${_options} --abbrev=0 "${current_branch}")
+
+        if(tag)
+            _falcosecurity_extract_version("${tag}" x y z match_found)
+
+            if(match_found)
+                math(EXPR z_plus_one "${z} + 1")
+                set(dev_version "${x}.${y}.${z_plus_one}")
```
Comment (Andreagit97) [inline cmake/modules/GetVersionFromGit.cmake:152]: uhm this means that on the same release branch, we can tag only patch versions for drivers tags `+driver`... it seems reasonable in the end we shouldn't introduce new changes in the schema in a release branch

**Review by Andreagit97 — COMMENTED** [review]
Great work, thank you!

File: cmake/modules/versions.cmake [hunk 2]
```diff
@@ -14,71 +14,10 @@
...
-    set(${_var}
-        "${dev_version}"
-        PARENT_SCOPE)
-    return()
-endfunction()
+include(GetVersionFromGit)
 
 function(get_libs_version _var)
-    _get_git_version(ver false)
+    get_version_from_git(ver "" "+driver")
```
[the remark this answers is not in the document]
Comment (leogr) [reply cmake/modules/versions.cmake:20]: Correct. It's somehow explained here :point_down: 

https://github.com/falcosecurity/libs/pull/1294/files#diff-869bda8f0ebe664f3168881fe2f057166a1f822ed6496303bf0d67f2a44b664fR104-R106

However, I agree to put a comment. I will fix it soon. Thank you

File: cmake/modules/GetVersionFromGit.cmake (hunk 1 above)
Comment (leogr) [reply cmake/modules/GetVersionFromGit.cmake:152]: Correct. This should also be documented in RELASE.md.

File: cmake/modules/versions.cmake (hunk 2 above)
[the remark this answers is not in the document]
Comment (leogr) [reply cmake/modules/versions.cmake:21]: done

File: cmake/modules/GetVersionFromGit.cmake (hunk 1 above)
Comment (leogr) [reply cmake/modules/GetVersionFromGit.cmake:160]: I've also updated RELEASE.md.

**Review by Andreagit97 — APPROVED** [review]
/approve

**Review by jasondellaluce — APPROVED** [review]
/approve

Love it ❤️

**Pull request merged**